core/tests/protractor_utils/ExplorationEditorFeedbackTab.js

```diff
@@ -56,122 +56,123 @@ var ExplorationEditorFeedbackTab = function() {
   /*
    * Workflows
    */
-  this.acceptSuggestion = function(suggestionDescription) {
-    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
-      var matchingSuggestionRows = rows.filter(function() {
-        return explorationFeedbackSubject.getText().then(function(subject) {
-          return suggestionDescription.indexOf(subject) !== -1;
-        });
-      });
-      expect(matchingSuggestionRows[1].isDisplayed()).toBe(true);
-      matchingSuggestionRows[1].click();
-      expect(viewSuggestionButton.isDisplayed()).toBe(true);
-      viewSuggestionButton.click();
-      expect(acceptSuggestionButton.isDisplayed()).toBe(true);
-      suggestionCommitMessageInput.sendKeys('Commit message');
-      acceptSuggestionButton.click();
-      waitFor.invisibilityOf(
-        acceptSuggestionButton, 'Suggestion modal takes too long to disappear');
-      waitFor.pageToFullyLoad();
-    });
+  this.acceptSuggestion = async function(suggestionDescription) {
+    var rows = await element.all(by.css(suggestionRowClassName));
+    var matchingSuggestionRows = [];
+    for (var i = 0; i < rows.length; i++) {
+      var subject = await rows[i].element(
+        by.css('.protractor-test-exploration-feedback-subject')).getText();
+      if (suggestionDescription.indexOf(subject) !== -1) {
+        matchingSuggestionRows.push(await rows[i]);
+      }
+    }
+    expect(await matchingSuggestionRows[0].isDisplayed()).toBe(true);
+    await matchingSuggestionRows[0].click();
+    expect(await viewSuggestionButton.isDisplayed()).toBe(true);
+    await viewSuggestionButton.click();
+    expect(await acceptSuggestionButton.isDisplayed()).toBe(true);
+    await suggestionCommitMessageInput.sendKeys('Commit message');
+    await acceptSuggestionButton.click();
+    await waitFor.invisibilityOf(
+      acceptSuggestionButton, 'Suggestion modal takes too long to disappear');
+    await waitFor.pageToFullyLoad();
   };
 
-  this.expectToHaveFeedbackThread = function() {
-    expect(feedbackTabRow.isPresent()).toBe(true);
+  this.expectToHaveFeedbackThread = async function() {
+    expect(await feedbackTabRow.isPresent()).toBe(true);
   };
 
-  this.getSuggestionThreads = function() {
+  this.getSuggestionThreads = async function() {
     var threads = [];
-    waitFor.visibilityOf(
+    await waitFor.visibilityOf(
       element.all(by.css(suggestionRowClassName)).first(),
       'No suggestion threads are visible');
-    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
-      rows.forEach(function() {
-        explorationFeedbackSubject.getText().then(function(subject) {
-          threads.push(subject);
-        });
-      });
-      return threads;
-    });
+    var rows = await element.all(by.css(suggestionRowClassName));
```

Comment:
ditto: Don't `await` the `element.all` call, and avoid using array methods below